src/api/controllers/tests/user-controller.test.ts

```diff
@@ -0,0 +1,230 @@
+import request from 'supertest';
+import app from '../../../app';
+import { userRepos } from '../../repos';
+
+describe("UserController", () => {
+    describe("GET /user", () => {
+        describe('getAllUsers endpoint', () => {
+            it("returns an non-empty array of users.", async () => {
+                //  Arrange
+                const endpoint = '/api/user/getAllUsers';
+                const expectedStatusCode = 200;
+                const userRepoSpy = jest
+                .spyOn(userRepos, 'getAllUsers')
+                .mockResolvedValueOnce([{
+                    firstName: 'Test',
+                    lastName: 'User',
+                    phone: '5141234567',
+                    email: 'test@gmail.com',
+                    firebase_id: 'testID',
+                    countryCode: 1,
+                    id: 1
+                }]);
+
+                //  Act
+                const res = await request(app).get(endpoint);
+                
+                //  Assert
+                expect(res.statusCode).toBe(expectedStatusCode);
+                expect(userRepoSpy).toHaveBeenCalledWith();
+                expect(userRepoSpy).toHaveBeenCalledTimes(1);
+                expect(res.headers['content-type']).toBe('application/json; charset=utf-8');
+                expect(res.body[0]).toHaveProperty('id');
+                expect(res.body[0]).toHaveProperty('firstName');
+                expect(res.body[0]).toHaveProperty('lastName');
+                expect(res.body[0]).toHaveProperty('countryCode');
+                expect(res.body[0]).toHaveProperty('phone');
+                expect(res.body[0]).toHaveProperty('email');
```

Comment:
```suggestion
                expect(userRepoSpy).toHaveBeenCalledWith();
                expect(userRepoSpy).toHaveBeenCalledTimes(1);
                expect(res).toEqual(expect.objectContaining({
                    statusCode: expectedStatusCode,
                    headers: expect.objectContaining({
                        'content-type': 'application/json; charset=utf-8',
                    }),
                    body: {
                        firstName: 'Test',
                        lastName: 'User',
                        phone: '5141234567',
                        email: 'test@gmail.com',
                        firebase_id: 'testID',
                        countryCode: 1,
                        id: 1,
                    },
                });
```

Reply:
Object can be verify in 1 expect function with expect.objectContaining({...}) and array can be verify with expect.arrayContaining([...]).  Those 2 are used when you want to check some of the value but not all as in the res object